1/src/components/counter/conteiner/CounterConteiner.js

```diff
@@ -0,0 +1,57 @@
+import {Component} from 'react';
+import { CounterView } from '../component/CounterView';
+
+class CounterContainer extends Component {
+    constructor(props) {
+        super(props)
+
+
+        this.state = {
+            countValue : 0,
+            isEven : true
+        }
+    }
+
+    componentDidUpdate() {
+        if(this.state.countValue % 2 === 0 && this.state.isEven !== true) {
+            this.setState({isEven : true})
+        } 
+        if (this.state.countValue % 2 !== 0 && this.state.isEven !== false) {
+            this.setState({isEven : false})
+        }
```

Comment:
it must be simplified. compare current state and previous state and then make a recalculation of parity